### ISSUE TITLE
Add ROWNUM() function

### ISSUE THREAD

**Issue by agershun** [agershun/alasql#95]
Suppose i have a table without a unique id. For example
```sql
    SELECT some_number, some_text FROM table;
```
some_number some_text
```
    42          "Test"
    84          "Another Test"
```
Now i want Alasql to additionally return the rownumber of every tuple:
```sql
    SELECT ROWNUM(), some_number, some_text FROM table;
```
```
    rownum some_number some_text
    0      42          "Test"
    1      84          "Another Test"
```
Do have a suggestion for me how to solve this issue?

**Comment by agershun** [comment]
Currently, there is no such function in Alasql.

The simplest way to solve this is to add a JavaScript lines after SELECT:
```js
    var res = alasql('SELECT some_number, some_text FROM tabl');
    res.forEach(function(d,idx){d.rownum = idx});
```

Unfortunately, ROWNUM() is not a function, but rather pseudo-function, because SQL server has to process it after ORDER BY clause.

**Comment by agershun** [comment]
T-SQL uses ROW_NUMBER() name and Orcale uses ROWNUM() for this functions.

**Comment by mathiasrw** [comment]
gosh.... would be so awesome if one could seæect the parser to match the favorite languarge flavour

**Comment by agershun** [comment]
For a now, these functions can be combined in one (it does not cost too much to have additional alias for this function). But for the future, it is a good idea to have them separate (may be to have the option flag like ```alasql.dialect = 'TSQL'```

**Comment by agershun** [comment]
Done. Will be available in version 0.1.6 soon or now in the develop branch.
```js
alasql('CREATE TABLE one (a INT PRIMARY KEY)');

for(var i=1;i<1000;i++) {
   alasql('INSERT INTO one VALUES (?)',[i]);
};

var res = alasql('SELECT * \
    FROM (SELECT a, ROWNUM() AS r FROM one)\
    WHERE r BETWEEN 55 AND 60');
```
You can try this example [in jsFiddle](http://jsfiddle.net/agershun/81noowmn/11/)

**Comment by abneralvarado** [comment]
Thanks!
